src/frontend/screens/SyncModal/index.js

```diff
@@ -50,53 +49,20 @@ const SyncModal = ({ navigation }: Props) => {
     },
   ] = React.useContext(ConfigContext);
 
-  const [listen, setListening] = React.useState<boolean>(false);
-  const [peers, syncPeer, syncGetPeers] = usePeers(listen, deviceName);
-  const [ssid, setSsid] = React.useState<null | string>(null);
+  const { ssid } = useWifiStatus();
+  const [peers, syncPeer, syncGetPeers] = usePeers(!!ssid, deviceName);
 
+  // Keep device awake on this screen
   React.useEffect(() => {
-    const subscriptions = [];
-    const handleConnectionChange = async (data: {}) => {
-      // NetInfoData does not actually tell us whether wifi is turned on, it just
-      // tells us what connection the phone is using for data. E.g. it could be
-      // connected to a wifi network but instead using 4g for data, in which case
-      // `data.type` will not be wifi. So instead we just use the event listener
-      // from NetInfo, and when the connection changes we look up the SSID to see
-      // whether the user is connected to a wifi network.
-      // TODO: We currently do not know whether wifi is turned off, we only know
-      // whether the user is connected to a wifi network or not.
-      let ssid = null;
-      try {
-        ssid = await NetworkInfo.getSSID();
-      } catch (e) {
-        bugsnag.notify(e);
-      } finally {
-        // Even if we don't get the SSID, we still want to show that a wifi
-        // network is connected.
-        setSsid(ssid);
-        // On connection change, ensure we have the latest peer info from
-        // backend.
-        syncGetPeers();
-        setListening(true);
-      }
-    };
-    // When the modal opens, start announcing this device as available for sync
-    setListening(true);
-    // Subscribe to NetInfo to know when the user connects/disconnects to wifi
-    subscriptions.push({
-      remove: NetInfo.addEventListener(handleConnectionChange),
-    });
-    // Keep the screen awake whilst on this screen
     KeepAwake.activate();
-    return () => {
-      // When the modal closes, stop announcing for sync
-      setListening(false);
-      // Unsubscribe all listeners
-      subscriptions.forEach(s => s.remove());
-      // Turn off keep screen awake
-      KeepAwake.deactivate();
-    };
-  }, [syncGetPeers]);
+    return KeepAwake.deactivate;
```

Comment:
Small nitpick, for the sake of a few extra characters it's always best to do `return () => KeepAwake.deactivate()` rather than this, for the same reason it's best not to pass a function directly to `myArray.map(library.method)`, because a non-breaking change could lead to an unexpected argument.

Reply:
ah yes good point! will fix